malcolm/modules/ADCore/util.py

```diff
@@ -76,6 +85,35 @@ def __init__(self,
         self.uniqueid = AUniqueIDArray(uniqueid)
 
 
+AttrSetTableElements = {
```

Comment:
Why not make a Table here and do TableMeta.from_table in the DetectorDriverPart? You will need to make Enums for sourceType, and dataType, but the one for AttributeDatasetType already exists.